File: Random/Distinct pairs.py
#//https://www.codechef.com/JAN19B/problems/DPAIRS
n,m=input().split()
n,m=int(n),int(m)
a= list(map(int,input().split()))
b= list(map(int,input().split()))
dic={}
# for i in range(len(a)):
#     for j in range(len(b)):
#         dic[a[i]+b[j]]=[i,j]


    
mina=a.index(min(a))
maxb=b.index(max(b))
if mina==maxb:
        maxb=b.index(min(b))

j=0
while (j<len(b) or j<len(a)):
        if j<len(b):
                dic[a[mina]+b[j]]=[mina,j]
        if j<len(a):
                dic[b[maxb]+a[j]]=[j,maxb]
        j+=1
j=0
print(dic)
for v in dic.values():
        if j<n+m-1:
                print(v[0],v[1])
        else:
                break
        j+=1

File: Random/Distinct Pairs.cpp
//https://www.codechef.com/JAN19B/problems/DPAIRS
#include<bits/stdc++.h>
using namespace std;
int main(){
    long int n,m;
    cin>>n>>m;
    vector<long long int> va,vb;
    for(int i=0;i<n;i++){
        long long tmp;
        cin>>tmp;
        va.push_back(tmp);
    }
    for(int i=0;i<m;i++){
        long long tmp;
        cin>>tmp;
        vb.push_back(tmp);
    }
    map<long long int,map<long,long>> dp;
    for(int i=0;i<n;i++){
        for(int j=0;j<m;j++){
            dp.insert(make_pair(va[i]+vb[j],make_pair(i,j)));
        }
    }
    for(auto i=dp.begin();i!=dp.end();i++){
        // for(auto j=data.begin();j!=data.end();j++){
            // cout<<"Sum-"<<i->first<<","<<j->first<<", "<<j->second<<endl;
            cout<<"Sum-"<<i->first<<" "<<i->second<<endl;
        // }
    }
    return 0;
}